mws/resource_mws_ncc_private_endpoint_rule.go

```diff
@@ -0,0 +1,69 @@
+package mws
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go/service/settings"
+	"github.com/databricks/terraform-provider-databricks/common"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceMwsNccPrivateEndpointRule() common.Resource {
+	s := common.StructToSchema(settings.NccAzurePrivateEndpointRule{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
+		for _, p := range []string{"network_connectivity_config_id", "group_id", "resource_id"} {
+			common.CustomizeSchemaPath(m, p).SetRequired().SetForceNew()
+		}
+		for _, p := range []string{"rule_id", "endpoint_name", "connection_state"} {
+			common.CustomizeSchemaPath(m, p).SetComputed()
+		}
+		return m
+	})
+	p := common.NewPairSeparatedID("network_connectivity_config_id", "rule_id", "/")
```

Comment:
Do we need to include account ID or workspace ID in this resource? I know it is included in some other resources.

Reply:
we do include it for `network_connectivity_config` resource, and since this resource depends on it - we should be safe. Packing 3 ids would require some changes to our packing logic